OpenRA.Game/Map/CellLayer.cs

```diff
@@ -34,15 +35,35 @@ public CellLayer(TileShape shape, Size size)
 			entries = new T[size.Width * size.Height];
 		}
 
-		public void CopyValuesFrom(CellLayer<T> anotherLayer)
+		// provides a default instance for this type of Layer
+		public static void Init(CellLayer<T> defaultLayer)
 		{
-			if (Size != anotherLayer.Size || Shape != anotherLayer.Shape)
+			defaultCellInfoLayer = defaultLayer;
+		}
+
+        public static void Init(Func<int, int, T> creator, int mapWidth, int mapHeight, TileShape tileShape)
```

Comment:
I did that following the Factory method pattern approach (previously it was even uglier, coupled inside the PathSearch). I could create a prototypemanager singleton class or a factory to delegate this method, but a static call inside cellLayer was simple, descriptive and effective IMHO.